### PR TITLE
Merged pnpm6 experiment into pnpm detector

### PR DESCRIPTION
**Pull request by pauld-msft**
`Pnpm6` has gone through a week or so of experimentation after being introduced with https://github.com/microsoft/component-detection/pull/1110 and has been validated to work as expected. 

This pull request:
- removes the experiment
- merges the pnpm6 changes into pnpm detector, using the suggested format from [this comment](https://github.com/microsoft/component-detection/pull/1110#discussion_r1602071025)
- increments Pnpm detector version from 5 -> 6

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmComponentDetectorFactory.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                this.Logger.LogWarning("Unsupported lockfileVersion in pnpm yaml file {File}", file.Location);
+            }
+            else
+            {
+                detector.RecordDependencyGraphFromFile(fileContent, singleFileComponentRecorder);
+            }
+        }
+        catch (Exception e)
+        {
```
Comment (cobya) [inline src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmComponentDetectorFactory.cs:66]: can you add a `FailedParsingFileRecord` telemetry object here? PipReportComponentDetector has an example

File: src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmComponentDetectorFactory.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            this.Logger.LogDebug("Skipping found file, it was detected as being within a {SkippedFolder} folder.", skippedFolder);
+        }
+
+        try
+        {
+            var fileContent = await new StreamReader(file.Stream).ReadToEndAsync();
+            var detector = this.GetPnpmComponentDetector(fileContent, file.Location);
+            if (detector == null)
+            {
```
Comment (cobya) [inline src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmComponentDetectorFactory.cs:57]: let's create a new telemetry item equivalent to `PipReportVersionTelemetryRecord` so we can track which versions aren't supported yet. I'm open to standardizing the telemetry object here as well so we can start applying it across our parsing logic

**Review by cobya — APPROVED** [review]
Changes look good to me 👍

File: src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmComponentDetectorFactory.cs (hunk 2 above)
Comment (pauld-msft) [reply src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmComponentDetectorFactory.cs:63]: created abstract VersionTelemetryRecord that both of these can inherit from

**Review by cobya — APPROVED** [review]

**Pull request merged**